buildSrc/src/main/java/com/google/firebase/gradle/ReleaseGenerator.kt

```diff
@@ -101,25 +205,60 @@ open class ReleaseGenerator : DefaultTask() {
   ) =
     repo
       .log()
-      .addPath("$directory/")
+      .addPath(directory)
       .addRange(previousReleaseRef, currentReleaseRef)
       .setMaxCount(1)
       .call()
       .iterator()
       .hasNext()
 
-  private fun writeReleaseConfig(configPath: File, libraries: List<String>, releaseName: String) {
-    File(configPath, "release.cfg")
-      .writeText(
-        """
-                    [release]
-                    name = $releaseName
-                    mode = RELEASE
-                    
-                    [modules]
-                    ${libraries.joinToString("\n".padEnd(21, ' '))}
-                """
-          .trimIndent()
-      )
+  private fun getDirChanges(
+    repo: Git,
+    previousReleaseRef: ObjectId,
+    currentReleaseRef: ObjectId,
+    directory: String
+  ) =
+    repo.log().addPath(directory).addRange(previousReleaseRef, currentReleaseRef).call().map {
+      toCommitDiff(repo, it)
+    }
+
+  private fun toCommitDiff(repo: Git, revCommit: RevCommit): CommitDiff {
+    return CommitDiff(
+      revCommit.id.name,
+      revCommit.authorIdent.name,
+      revCommit.fullMessage,
+    )
+  }
+
+  private fun writeReleaseReport(file: File, report: String) {
+    file.writeText(report)
+  }
+
+  private fun writeReleaseConfig(file: File, config: ReleaseConfig) {
+    file.writeText(config.toFile())
+  }
+
+  private fun getRelativeDir(project: Project) = project.path.substring(1).replace(':', '/')
+}
+
+data class ReleaseConfig(val releaseName: String, val libs: Set<String>) {
+  companion object {
+    fun fromFile(file: File): ReleaseConfig {
+      val contents = file.readText(Charsets.UTF_8).split("\n")
```

Comment:
Kotlin has a method for this actually:
```kotlin
file.readLines()
```

Also note that `Charsets.UTF_8` is the default.